LibGAP.jl/src/gap_to_julia.jl

```diff
@@ -0,0 +1,186 @@
+## Converters
+
+## Default
+gap_to_julia(::Type{GAPInputType},x::GAPInputType, recusive_dict = nothing) = x
+gap_to_julia(::Type{Any},         x::GAPInputType, recusive_dict = nothing) = gap_to_julia(x)
+gap_to_julia(::Type{Any},         x::Any         , recusive_dict = nothing) = x
+
+## Integers
+gap_to_julia(::Type{Int128} ,x::Int64, recusive_dict = nothing) = trunc(Int128 ,x)
+gap_to_julia(::Type{Int64}  ,x::Int64, recusive_dict = nothing) = x
+gap_to_julia(::Type{Int32}  ,x::Int64, recusive_dict = nothing) = trunc(Int32  ,x)
+gap_to_julia(::Type{Int16}  ,x::Int64, recusive_dict = nothing) = trunc(Int16  ,x)
+gap_to_julia(::Type{Int8}   ,x::Int64, recusive_dict = nothing) = trunc(Int8   ,x)
+
+## Unsigned Integers
+gap_to_julia(::Type{UInt128},x::Int64, recusive_dict = nothing) = trunc(UInt128,x)
+gap_to_julia(::Type{UInt64} ,x::Int64, recusive_dict = nothing) = trunc(UInt64 ,x)
+gap_to_julia(::Type{UInt32} ,x::Int64, recusive_dict = nothing) = trunc(UInt32 ,x)
+gap_to_julia(::Type{UInt16} ,x::Int64, recusive_dict = nothing) = trunc(UInt16 ,x)
+gap_to_julia(::Type{UInt8}  ,x::Int64, recusive_dict = nothing) = trunc(UInt8  ,x)
+
+## BigInts
+gap_to_julia(::Type{BigInt}, x::Int64, recusive_dict = nothing) = BigInt( x )
+
+function gap_to_julia(::Type{BigInt}, x::MPtr, recusive_dict = nothing)
+    ## Check for correct type
+    if ! Globals.IsInt(x)
+        throw(ArgumentError("GAP object is not a large integer"))
+    end
+    ## get size of GAP BigInt (in limbs), multiply
+    ## by 64 to get bits
+    size_limbs = ccall(:GAP_SizeInt,Cint,(MPtr,),x)
+    size = abs(size_limbs * sizeof(UInt) * 8)
+    ## allocate new GMP
+    new_bigint = Base.GMP.MPZ.realloc2(size)
+    new_bigint.size = size_limbs
+    ## Get limb address ptr
+    addr = ccall(:GAP_AddrInt,Ptr{UInt},(MPtr,),x)
+    ## Copy limbs
+    unsafe_copyto!( new_bigint.d, addr, abs(size_limbs) )
+    return new_bigint
+end
+
+## Rationals
+function gap_to_julia(::Type{Rational{T}}, x::Int64, recusive_dict = nothing) where T <: Integer
+    numerator = gap_to_julia(T,x)
+    return numerator // T(1)
+end
+
+function gap_to_julia(::Type{Rational{T}}, x::MPtr, recusive_dict = nothing) where T <: Integer
```

Comment:
According to Codecov, there are no tests for this function at all, it is never called :-( Same for the other rationals conversion above, and the float conversion below, and more -- see also https://codecov.io/gh/oscar-system/GAPJulia/pull/160/diff

Reply:
(side remark: is `recusive_dict` a typo?)